### PR TITLE
argo 2.9.3 (new formula)

### PR DESCRIPTION
**Pull request by crunchtime-ali**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This is WIP. I am still getting the following errors and hope that someone can help me fix thme:

> go build net: copying /Users/me/Library/Caches/Homebrew/go_cache/8f/8fab70c85648629b1f00a730a1c0fd7a780692d0f76b6416d602443640bba227-d: open /usr/local/Cellar/go/1.14.5/libexec/pkg/darwin_amd64/net.a: operation not permitted
> go build os/user: copying /Users/me/Library/Caches/Homebrew/go_cache/01/01cec212fc74e99b9a524eccac5e049f251b8fbcda0ee071d10675b065dc967f-d: open /usr/local/Cellar/go/1.14.5/libexec/pkg/darwin_amd64/os/user.a: operation not permitted

### CODE REVIEW

**Comment by crunchtime-ali** [comment]
I am also getting the `operation not permitted error for `user.a` and `net.a` in CI. Does anyone have a clue how I can fix that?

https://github.com/Homebrew/homebrew-core/pull/58259/checks?check_run_id=888857864#step:6:277
https://github.com/Homebrew/homebrew-core/pull/58259/checks?check_run_id=888857864#step:6:297

> go build os/user: copying /private/tmp/go-build946619233/b142/_pkg_.a: open /usr/local/Cellar/go/1.14.5/libexec/pkg/darwin_amd64/os/user.a: operation not permitted
> go build net: copying /private/tmp/go-build946619233/b074/_pkg_.a: open /usr/local/Cellar/go/1.14.5/libexec/pkg/darwin_amd64/net.a: operation not permitted

File: Formula/argo.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      :tag      => "v2.9.3",
+      :revision => "9407e19b3a1c61ad4043e382484fd0b6b15574f2"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
```
Comment (SMillerDev) [inline Formula/argo.rb:14]: Does this not work with go modules?

Comment (crunchtime-ali) [reply Formula/argo.rb]: Yes, after removing `GOPATH` etc. the same problem remains.

**Comment by SMillerDev** [comment]
I don't see those errors right now. It just seems to fail silently

**Comment by crunchtime-ali** [comment]
> I don't see those errors right now. It just seems to fail silently

They are still there in the CI logs, for example here https://github.com/Homebrew/homebrew-core/pull/58259/checks?check_run_id=889596249#step:6:191

I was able to work around the problem by removing `CGO_ENABLED=0` and am working on a test-case now.

**Comment by crunchtime-ali** [comment]
It works now and I also added some simple tests @SMillerDev

File: Formula/argo.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  def install
+    inreplace "Makefile", "CGO_ENABLED=0", ""
+    system "make", "dist/argo-darwin-amd64"
+    bin.install "dist/argo-darwin-amd64" => "argo"
+    prefix.install_metafiles
```
Comment (dawidd6) [inline Formula/argo.rb:17]:
```suggestion
```

File: Formula/argo.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      :revision => "9407e19b3a1c61ad4043e382484fd0b6b15574f2"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  def install
+    inreplace "Makefile", "CGO_ENABLED=0", ""
+    system "make", "dist/argo-darwin-amd64"
```
Comment (dawidd6) [inline Formula/argo.rb:15]: Could we somehow make this step more OS-agnostic? I mean, this will probably fail on Linux.

Comment (crunchtime-ali) [reply Formula/argo.rb]: @dawidd6 I must have overlooked the more general target `dist/argo` which I now use.

File: Formula/argo.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      :tag      => "v2.9.3",
+      :revision => "9407e19b3a1c61ad4043e382484fd0b6b15574f2"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+  depends_on "yarn" => :build
+
+  def install
+    inreplace "Makefile", "CGO_ENABLED=0", ""
```
Comment (chenrui333) [inline Formula/argo.rb:14]: same here, would be good to add some comments for this change for future reference.

Comment (crunchtime-ali) [reply Formula/argo.rb:15]: @chenrui333 I added a comment about it

**Comment by crunchtime-ali** [comment]
Can someone please kick the CI server @chenrui333 ?

**Review by SMillerDev — APPROVED** [review]
Thanks @crunchtime-ali  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @crunchtime-ali!!

**Pull request closed without merging**